botocore/exceptions.py

```diff
@@ -775,3 +775,27 @@ class FlexibleChecksumError(BotoCoreError):
 
 class InvalidEndpointConfigurationError(BotoCoreError):
     fmt = 'Invalid endpoint configuration: {msg}'
+
+
+class EndpointRuleSetError(BotoCoreError):
+    """Base error for endpoint ruleset errors"""
+
+    fmt = '{msg}'
+
+
+class EndpointResolutionError(EndpointRuleSetError):
+    """Error when an issue arises in resolving an endpoint"""
+
+    fmt = '{msg}'
```

Comment:
The ones that don't have anything specific in the message probably don't need `{msg}`. We can use pass like the examples above.

Reply:
Adding a nit to this: That blank line between docstring and code isn't needed, but black won't auto-remove it.